src/map/skill.cpp

```diff
@@ -7574,6 +7574,78 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 
 		break;
 
+	case NPC_IMMUNE_PROPERTY:
+		switch (skill_lv) {
+			case 1:
+				type = SC_IMMUNE_PROPERTY_NOTHING;
+				break;
+			case 2:
+				type = SC_IMMUNE_PROPERTY_WATER;
+				break;
+			case 3:
+				type = SC_IMMUNE_PROPERTY_GROUND;
+				break;
+			case 4:
+				type = SC_IMMUNE_PROPERTY_FIRE;
+				break;
+			case 5:
+				type = SC_IMMUNE_PROPERTY_WIND;
+				break;
+			case 6:
+				type = SC_IMMUNE_PROPERTY_DARKNESS;
+				break;
+			case 7:
+				type = SC_IMMUNE_PROPERTY_SAINT;
+				break;
+			case 8:
+				type = SC_IMMUNE_PROPERTY_POISON;
+				break;
+			case 9:
+				type = SC_IMMUNE_PROPERTY_TELEKINESIS;
+				break;
+			case 10:
+				type = SC_IMMUNE_PROPERTY_UNDEAD;
+				break;
+		}
+		clif_skill_nodamage(src,bl, skill_id,skill_lv,sc_start(src,bl,type,100,skill_lv,skill_get_time(skill_id,skill_lv)));
+		sc_start(src,bl, SC_IMMUNE_PROPERTY,100,skill_lv,skill_get_time(skill_id,skill_lv));
+		break;
+
+	case NPC_MOVE_COORDINATE:
+		{
+			int16 x, y;
+			uint8 dir = map_calc_dir(bl, src->x, src->y);
+
+			if (dir > 0 && dir < 4)
+				x = 1;
+			else if (dir > 4)
+				x = -1;
+			else
+				x = 0;
+			if (dir > 2 && dir < 6)
+				y = 1;
+			else if (dir == 7 || dir < 2)
+				y = -1;
+			else
+				y = 0;
+			if (status_get_class_(src) == CLASS_BOSS) {//if caster is a boss, the skill will pull target ahead of the boss, instead of switch coordinates
+				if (skill_check_unit_movepos(1, bl, src->x, src->y, 1, 1)) {
+					clif_blown(src);
+				}
+			}
+			else {
+				if (skill_check_unit_movepos(1, bl, src->x, src->y, 1, 1)) {
+					clif_blown(bl);
+				}
+				if (unit_movepos(src, bl->x, bl->y, 1, 1)) {
+					clif_blown(src);
+				}
+			}
```

Comment:
```suggestion
			if (skill_check_unit_movepos(1, bl, src->x, src->y, 1, 1)) {
				clif_blown(bl);
			}
			if (status_get_class_(src) != CLASS_BOSS && unit_movepos(src, bl->x, bl->y, 1, 1)) { // If caster is not a boss, the skill will pull target ahead of the caster, instead of switching coordinates
				clif_blown(src);
			}
```

Reply:
Sorry my mistake. This part should not be same but I missed. I fixed it with another small fix.